packages/forms/src/Components/Builder.php

```diff
@@ -831,6 +831,23 @@ public function isBlockLabelTruncated(): bool
         return (bool) $this->evaluate($this->isBlockLabelTruncated);
     }
 
+    /**
+     * @return Component[]
+     */    
+    public function getBlockPickerBlocks(): array
+    {
+        $state = $this->getState();
+    
+        return array_filter($this->getBlocks(), function ($block) use ($state) {
+            /** @var Block $block */
+            $count = collect($state)->filter(function ($item) use ($block) {
+                return $item['type'] === $block->getName();
+            })->count();
+
+            return is_null($block->getMaxItems()) || $count < $block->getMaxItems();
+        });
+    }
```

Comment:
If there is no maximum, please prevent calculating the number of existing blocks